docs/concepts/renamer.md

```diff
@@ -0,0 +1,251 @@
+
+Floating point numbers are transformed into strings, with two places after
+the comma.  `1.2345` is transformed into `1.23`, `1` is transformed into
+`1.00`.  The representation is not locale aware, meaning that we don't
+use `,` as the decimal delimiter in countries such as Germany.
+
+For seldom cases that a floating point number is NaN (not a number) or
++/- Infinity, we use `NaN` and `inf`/`-inf`.
+
+
+#### Index Operator
+
+The index operator `<list[N]>` can be used to access the N-th value of `list`.
+The variable `list` must exist, but if the list does not have N entries,
+an empty string is used instead.
+
+The operator can also be used to access characters of strings or to create
+substrings.  This is similar to how Python (NumPy) handles slices.
+`<str[0]>` returns the first character of string `str` or an empty string if
+`str` is empty itself.  `<str[:3]>` returns the first three characters of `str`
+and is equivalent to `<str[0]><str[1]><str[2]>`.
+`[:3]` is short for `[0:3]`, meaning that we want the character subrange
+`[0,2)`.
+
+The index operator can't be used on numbers (integers and floats).
+
+
+#### Number Format
+
+For filenames, we often want numbers that have N digits, e.g. via leading
+zeroes.  For floats, we sometimes want at most / minimum two decimal places.
+
+For integers, we support (example number `1234`):
+
+```
+{field}     1234
+{field:6}   001234
+{field:06}  000124
+{field:2}   1234
+```
+
+Syntax: `{variable:min_digits}` and `{variable:0min_digits}`, meaning that
+the number `6` in above example means "at least six digits".  An optional
+`0` can be used to make it more clear that `0` is used as a filler.
+This could also prove useful if we ever want to allow other filler characters.
+
+
+For floating point numbers, we support (example number `123.34567`):
+
+```
+{field}     123.35
+{field:6}   000123.35
+{field:06}  000123.35
+{field:6,3} 000123.346
+{field:6,8} 000123.34567000
+{field:2,3} 123.346
+```
+
+Syntax:
+ - `{variable:min_digits}`
+ - `{variable:0min_digits}`
+ - `{variable:min_digits,decimal_digits}`
+ - `{variable:0min_digits,decimal_digits}`
+
+`min_digits` meaning the same as for integers. `decimal_digits` refers to the
+number of digits after the decimal dot.
+
+Default: two decimal places.
+
+If ever required, we could also introduce number signs (`+`/`-`), but I'm not
+aware of negative numbers in MediaElch except some default values for
+"display season".
+
+
+#### Field Access
+
+Some fields are themselves structured.  For example, if we want to use an
+actors name in a filename, we could use `<actors[0].name>`.  For the actor's
+character, one could use `<actors[0].role>`.
+
+In this example, `actors` is a list (see "Index Operator"), and its elements
+are "actor" _structures_.
+
+The `.` allows access into a structure's fields.
+
+Imagine that a movie had a structured field `rating` that is not a list.
+If we want to access the count of votes, we could use `rating.voteCount`.
+
```

Comment:
How can you handle genres ?

For exemples with IMDB scrapper :
- "Gladiator" will have genres "Action, Adventure, Drama", sort by alphabetical order. So `<genres[0]>` will give a "Action" result.
- "Black Hawk Down" will have genres "Action, Drama, History", sort by alphabetical order. So `<genres[0]>` will give a "Action" result.
- "Pacific Rim" will have genres "Action, Adventure, Sci-Fi", sort by alphabetical order. So `<genres[0]>` will give a "Action" result.
- "Die Hard"  will have genres "Action, Thriller", sort by alphabetical order. So `<genres[0]>` will give a "Action" result.

Each of these films would be better defined by any other genre than the very generic "Action".
Perhaps it would be interesting to allow genres to be reordered by film to put the most relevant first?
It will be the same on the tags.

Reply:
Hi, sorting is a requirement for this to work. It was suggested in https://github.com/Komet/MediaElch/issues/1575 :)
Since Kodi's NFO file's use simple XML tags for genres, we can sort them internally according to the NFO file. A next step would be to allow sorting genres via the UI.

I have added a sentence or two about that to this document.